### PR TITLE
Smaller spine library, save 14KB package size

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #17715

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: cocos/spine/lib/spine-define.ts [hunk 1]
```diff
@@ -106,486 +106,149 @@ function overrideClass (wasm): void {
...
-            proto: prototype,
-            property: 'isEmpty',
-            getter: prototype.isEmpty,
-        },
-        {
-            proto: prototype,
-            property: 'str',
-            getter: prototype.str,
-        },
+        ['length', prototype.length],
```
Comment (minggo) [inline cocos/spine/lib/spine-define.ts:120]: Why can change object to array?

Comment (dumganhar) [reply cocos/spine/lib/spine-define.ts:120]: For smaller code.  You could compare the code, and using array will make the code cleaner.

**Review by bofeng-song — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
Windows PR test result:
https://cctest.cocos.org/#/reportview/PR-TEST/a4680461-8e51-496e-a0e7-a35e962240a3/-1~undefined

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Pull request merged**